Random Doubles: Movepool updates (Gen 1-5)

n10sit opened the pull request.

TheImmortal commented:
@ascriptmaster can you review

ascriptmaster commented on data/formats-data.js:
```diff
@@ -1542,12 +1542,12 @@ exports.BattleFormatsData = {
 	porygonz: {
 		randomBattleMoves: ["triattack", "shadowball", "icebeam", "thunderbolt", "conversion", "trick", "nastyplot"],
-		randomDoubleBattleMoves: ["protect", "triattack", "darkpulse", "hiddenpowerfighting", "agility", "trick", "nastyplot"],
+		randomDoubleBattleMoves: ["protect", "triattack", "darkpulse", "hiddenpowerfighting", "ice beam", "thunderbolt", "agility", "trick", "nastyplot"],
```
no space in `icebeam`

ascriptmaster commented on data/formats-data.js:
```diff
@@ -3001,7 +3007,7 @@ exports.BattleFormatsData = {
 	flygon: {
 		randomBattleMoves: ["earthquake", "outrage", "uturn", "roost", "stoneedge", "firepunch", "dragondance"],
-		randomDoubleBattleMoves: ["earthquake", "protect", "dragonclaw", "uturn", "rockslide", "firepunch", "fireblast", "tailwind", "feint"],
+		randomDoubleBattleMoves: ["earthquake", "protect", "dragonclaw", "uturn", "rockslide", "firepunch", "fireblast", "tailwind", "feint", "dragondance"],
```
Actually, Feint off a base 100 without STAB or an ability to boost it is not very great. Remove it here

ascriptmaster commented on data/formats-data.js:
```diff
@@ -3765,7 +3772,7 @@ exports.BattleFormatsData = {
 	wormadamtrash: {
 		randomBattleMoves: ["stealthrock", "toxic", "gyroball", "protect"],
-		randomDoubleBattleMoves: ["strugglebug", "stringshot", "gyroball", "protect"],
+		randomDoubleBattleMoves: ["strugglebug", "stringshot", "gyroball", "bug buzz", "flash cannon", "suckerpunch", "protect"],
```
remove spaces in moves here: `bugbuzz`, `flashcannon`

ascriptmaster commented on data/formats-data.js:
```diff
@@ -3870,7 +3877,7 @@ exports.BattleFormatsData = {
 	bronzong: {
 		randomBattleMoves: ["stealthrock", "earthquake", "toxic", "reflect", "lightscreen", "trickroom", "explosion", "gyroball"],
-		randomDoubleBattleMoves: ["earthquake", "protect", "reflect", "lightscreen", "trickroom", "explosion", "gyroball"],
+		randomDoubleBattleMoves: ["earthquake", "protect", "reflect", "lightscreen", "trickroom", "explosion", "gyroball", "zenheadbutt"],
```
Zen is honestly pretty meh on the zong. Any specific reason why it's added?

ascriptmaster commented on data/formats-data.js:
```diff
@@ -4545,7 +4552,7 @@ exports.BattleFormatsData = {
 	audino: {
 		randomBattleMoves: ["wish", "protect", "healbell", "toxic", "thunderwave", "reflect", "lightscreen", "doubleedge"],
-		randomDoubleBattleMoves: ["healpulse", "protect", "healbell", "trickroom", "thunderwave", "reflect", "lightscreen", "doubleedge", "helpinghand"],
+		randomDoubleBattleMoves: ["healpulse", "protect", "healbell", "trickroom", "thunderwave", "reflect", "lightscreen", "doubleedge", "helpinghand", "hypervoice", "dazzlinggleam"],
```
I don't think regular Audino, which is pure Normal, needs Dazzling Gleam for any reason.

ascriptmaster commented on data/formats-data.js:
```diff
@@ -4662,7 +4669,7 @@ exports.BattleFormatsData = {
 	krookodile: {
 		randomBattleMoves: ["earthquake", "stoneedge", "pursuit", "knockoff", "stealthrock", "superpower"],
-		randomDoubleBattleMoves: ["earthquake", "stoneedge", "protect", "knockoff", "superpower"],
+		randomDoubleBattleMoves: ["earthquake", "stoneedge", "protect", "knockoff", "superpower", "firefang"],
```
Why does Krook need Fire coverage? Its other moves cover it well enough and it doesn't even have Strong Jaw

ascriptmaster commented on data/formats-data.js:
```diff
@@ -828,7 +828,7 @@ exports.BattleFormatsData = {
 	golemalola: {
 		randomBattleMoves: ["stealthrock", "stoneedge", "thunderpunch", "earthquake", "toxic"],
-		randomDoubleBattleMoves: ["doubleedge", "stoneedge", "rockslide", "protect"],
+		randomDoubleBattleMoves: ["doubleedge", "stoneedge", "rockslide", "thunderpunch", "earthquake", "protect"],
```
Has this been rolling without Galvanize? Because looking at the scripts file, that's a bigger issue and Thunderpunch isn't needed if it can rely on Double Edge for the Electric STAB.

ascriptmaster commented:
Why are these listed as movepool updates for Gen 1-5 when you're editing the Gen 7 file? Plus the obvious fact that there was no doubles format until Gen 3 anyways

n10sit commented:
Thanks for the check, will fix what I messed up, as for your questions: I assumed you could get non-galvanize Golem-Alola since you could get it in Singles, so I added t punch. Zen on Bronzong was just to give it another attack to try and avoid those 4 status move sets, which is also why I added attacks on Audino (I forgot it was pure normal though so I'll drop gleam). And I described the updates as gens 1 through 5 since the Pokemon I went through were from gens 1 through 5, but I could see how that could be confusing, should have just left it as "movepool updates"

ascriptmaster commented:
A-Golem can have non-Galvanize sets. However, a change to `scripts.js` should likely be done to make sure that Double-Edge becomes "paired" with Galvanize (abilities are always chosen after the moves have been selected)

TheImmortal commented:
I don't see Bronzong would get 4 status moves. Gyro Ball shouldn't ever get rejected on it.

AustinXII commented:
You probably should have renamed the commit

AustinXII commented:
@TheImmortal
